__tests__/database.test.js

```diff
@@ -59,13 +58,120 @@ describe("GET endpoints", () => {
     });
 });
 
+
+describe("GET articles", () => {
+    test("Request from /api/articles/1 returns an object with required values", () => {
+        const matchArticleObject = {
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String)
+          }
+        return request(app)
+        .get('/api/articles/1')
+        .expect(200)
+        .then(({body}) => {
+            expect(body).toMatchObject(matchArticleObject)
+        })
+    })
+    test("Request from /api/articles/:id with a valid but missing id responds with 404 article not found", () => {
+        return request(app)
+        .get('/api/articles/999999')
+        .expect(404)
+        .then(({body}) => {
+            expect(body.message).toBe('article not found')
+        })
+    })
+    test("Request from /api/articles/invalid responds with 400 Bad request", () => {
+        return request(app)
+        .get('/api/articles/invalid')
+        .expect(400)
+        .then(({body}) => {
+            expect(body.message).toBe('Bad request')
+        })
+    })
+    test("Request from /api/articles returns an array of all articles in descending date order", () => {
+        const matchArticleObject = {
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(Number)
+          }
+        return request(app)
+        .get('/api/articles')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.length).toBe(13)
+            body.forEach(article => {
+                expect(article).toMatchObject(matchArticleObject)
+                expect(Object.keys(article).includes('body')).toBeFalsy()
+            });
+            expect(body).toBeSorted({key: 'created_at', descending: true})
+        })
+    })
+})
+
+describe("GET comments", () => {
+    test("Request from /api/articles/1/comments returns an array of comments for the specified article", () => {
+        const matchCommentObject = {
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: 1
+          }
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.length).toBe(11)
+            body.forEach(article => {
+                expect(article).toMatchObject(matchCommentObject)
+            });
```

Comment:
Great test here and well done for recognising that we can be and therefore should be specific with the article_id but generic with the rest 👏